packages/@aws-cdk/aws-servicecatalog/lib/private/association-manager.ts

```diff
@@ -74,6 +74,28 @@ export class AssociationManager {
     }
   }
 
+  public static constrainProvisioningParameters(portfolio: IPortfolio, product: IProduct,
+    assertion: TemplateRule, options: CommonConstraintOptions): void {
+    this.validateCommonConstraintOptions(portfolio, product, options);
+    const association = this.associateProductWithPortfolio(portfolio, product);
+    const constructId = `LaunchTemplateConstraint${hashValues(association.associationKey, assertion.ruleName)}`;
+
+    if (!portfolio.node.tryFindChild(constructId)) {
+      const constraint = new CfnLaunchTemplateConstraint(portfolio as unknown as cdk.Resource, constructId, {
+        acceptLanguage: options.messageLanguage,
+        description: options.description,
+        portfolioId: portfolio.portfolioId,
+        productId: product.productId,
+        rules: this.formatRule(portfolio as unknown as cdk.Resource, assertion),
+      });
+
+      // Add dependsOn to force proper order in deployment.
+      constraint.addDependsOn(association.cfnPortfolioProductAssociation);
+    } else {
+      throw new Error(`Provisioning rule ${assertion.ruleName} already configured on association ${this.prettyPrintAssociation(portfolio, product)}`);
```

Comment:
There's some thought to making this override the previous Rule, instead of erroring out.

Maybe an option in `constrainProvisioningParameters`?

Reply:
This is here because in CFN templates you would have all the provisioning rules listed in the same construct, so you can't have overlapping names which are the keys in the map.  I broke it apart here like the event notifications, since one would probably only ever add 1-2 rules per product, but I think we still would rather keep unique names in case we need to shift back to that mode of adding all rules in one construct, which would be less of a breaking change if we enforce unique names now.